plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1046,7 +1055,7 @@ private void finishOptimize(ConnectorSession session, IcebergTableExecuteHandle
         }
 
         // try to leave as little garbage as possible behind
-        if (optimizeHandle.isRetriesEnabled()) {
+        if (sourceHandle.getRetryMode() != NO_RETRIES) {
```

Comment:
are we setting retryMode in the source handle. We probably should, but I skimmed the code and it looks it is alway NO_RETRIES in the OPTIMIZE flow

Reply:
> it is alway NO_RETRIES in the OPTIMIZE flow

good catch. that's error prone.
should we pass RetryMode to beginTableExecute?

Reply:
This is passing the same information twice in the flow. But I guess this is fine.